Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/node/FunctionsNode.java

```diff
@@ -28,18 +32,26 @@ public FunctionsNode(@Nonnull FunctionApp functionApp) {
         this.withIcon(AzureIcons.FunctionApp.MODULE)
             .withLabel("Functions")
             .withActions(FunctionAppActionsContributor.FUNCTIONS_ACTIONS)
-            .addChildren(a -> a.listFunctions().stream().sorted(Comparator.comparing(FunctionEntity::getName)).collect(Collectors.toList()), (a, functionsNode) -> new Node<>(a)
+            .addChildren(this::listChildren, (a, functionsNode) -> new Node<>(a)
                 .withIcon("/icons/function-trigger.png")
                 .withLabel(FunctionEntity::getName)
                 .withDescription(FunctionsNode::getFunctionTriggerType)
                 .onDoubleClicked(FunctionAppActionsContributor.TRIGGER_FUNCTION)
                 .withActions(FunctionAppActionsContributor.FUNCTION_ACTION));
 
         this.listener = new AzureEventBus.EventListener(this::onEvent);
-        AzureEventBus.on("resource.refreshed.resource", listener);
         AzureEventBus.on("appservice|function.functions.refresh", listener);
     }
 
+    private List<FunctionEntity> listChildren(FunctionApp functionApp) {
```

Comment:
did you figure out why it will list the triggers?

Reply:
Before we listen to event "resource.refreshed.resource" in Functions node, and when we list function apps, it will trigger resource refreshed event after runtime loaded, which trigger Functions node to reload children.

So just remove the listener on resource refreshed in functions node